### ISSUE TITLE
Logs are not forwareded during VM (re)creation for some time

### ISSUE THREAD

**Issue by johha** [cloudfoundry/syslog-release#134]
While (re)creating a bosh VM log messages are not forwarded (e.g. to ELK) for some time.
Can be reproduced by recreating a e.g. a CF API VM:

#### Bosh Recreate

```bosh -d cf recreate api/c4a48fed-a833-4dc7-8505-ea222fd92881
Using environment 'https://bosh.internal:25555/' as user 'admin'

Using deployment 'cf'

Continue? [yN]: y

Task 9444101

Task 9444101 | 08:28:22 | Preparing deployment: Preparing deployment (00:00:02)
Task 9444101 | 08:28:24 | Preparing deployment: Rendering templates (00:00:28)
Task 9444101 | 08:28:53 | Preparing package compilation: Finding packages to compile (00:00:00)
Task 9444101 | 08:28:57 | Updating instance api: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:28:57 | L executing pre-stop: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:28:58 | L executing drain: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:29:08 | L stopping jobs: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:29:29 | L executing post-stop: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:31:47 | L installing packages: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:32:20 | L configuring jobs: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:32:21 | L executing pre-start: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:32:39 | L starting jobs: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary)
Task 9444101 | 08:33:24 | L executing post-start: api/c4a48fed-a833-4dc7-8505-ea222fd92881 (3) (canary) (00:04:28)

Task 9444101 Started  Thu May 11 08:28:22 UTC 2023
Task 9444101 Finished Thu May 11 08:33:25 UTC 2023
Task 9444101 Duration 00:05:03
Task 9444101 done
```

#### Not forwarded BUT existing CC log message
```json
{
  "timestamp": "2023-05-11T08:32:42.397340946Z",
  "message": "database schema is as new or newer than locally available migrations",
  "log_level": "info",
  "source": "cc.db.wait_until_current",
  "data": {},
  "thread_id": 89640,
  "fiber_id": 89660,
  "process_id": 8,
  "file": "/var/vcap/data/packages/cloud_controller_ng/2f04b7e748ae77c6e82851873e740d2f94d86a42/cloud_controller_ng/lib/cloud_controller/db_migrator.rb",
  "lineno": 43,
  "method": "wait_for_migrations!"
}
```

#### First forwarded log message
```json
{
  "timestamp": "2023-05-11T08:32:59.532481554Z",
  "message": "(0.001351s) (app/controllers/v3/builds_controller.rb:100:in `show') (query_length=91) SELECT * FROM \"apps\" WHERE (\"apps\".\"guid\" = '641af42f-d006-4d3a-9270-0cc73c6f3871') LIMIT 1",
  "log_level": "info",
  "source": "cc.db",
  "data": {
    "request_guid": "1238a46e-f843-4fb4-717e-36522725229d::b93672f5-6959-48c6-8c49-896e3e499ec6",
    "user_guid": "b2f35cb1-c645-4d33-a0f6-24211ff32171",
    "b3_trace_id": "3ba144d7599d91aa66ab88f73d837fd8",
    "b3_span_id": "66ab88f73d837fd8"
  },
  "thread_id": 105440,
  "fiber_id": 105460,
  "process_id": 8,
  "file": "/var/vcap/data/packages/cloud_controller_ng/2f04b7e748ae77c6e82851873e740d2f94d86a42/gem_home/ruby/3.2.0/gems/sequel-5.66.0/lib/sequel/database/logging.rb",
  "lineno": 88,
  "method": "public_send"
}
```
**->** Logs are not forwarded for about 10 seconds and missing in ELK.  Issue should be reproducible by recreating any Bosh VM and checking its logs.
Could this be an issue with a wrong/missing health check or the bosh process startup order?

**Comment by johha** [comment]
Is this related to #131?

**Comment by KarstenSchnitter** [comment]
The default setup with Blackbox and the syslog-forwarder is unable to fully support the use-case. First, Blackbox is only tailing log files. Any messages are basically dropped, when Blackbox is not running. Since Blackbox is a common bosh job, it cannot pick up any lifecycle logs outside its own lifecycle. 

Internally at SAP, I have replaced Blackbox with Fluent Bit using its Tail DB to continue reading files, after a restart. This allows to read all log messages, unless a VM is deleted. @johha: I am happy to share the configuration for this, if you are interested.

There is one more issue with the syslog-forwarded explained in #131. This relates to the problem, that the syslog-forwarder is stopped in the drain script. That way it is stopped even before Blackbox and other jobs are stopped. This is addressed with #133.

**Comment by ctlong** [comment]
Closing in favor of #71, since I think that would fix this issue given that syslog would start forwarding component logs at the pre-start step.

If that turns out not to work then we can always re-open this issue.
